scipy/special/_trig.h

```diff
@@ -0,0 +1,114 @@
+/* Implement sin(pi*z) and cos(pi*z) for complex z. Since the periods
+ * of these functions are integral (and thus better representable in
+ * floating point), it's possible to compute them with greater accuracy
+ * than sin(z), cos(z).
+ *
+ * Originally written in Cython by Josh Wilson (github @person142) 2016. 
+ * Translated to C++ by SciPy developers in 2023.
+ */
+
+
+#pragma once
+
+
+#include <cmath>
+#include <complex>
+#include <limits>
+
+#include "cephes.hh"
+#include "special/evalpoly.h"
+
+
+
+namespace special {
+	
+
+    inline std::complex<double> sinpi(std::complex<double> z) {
+	double x = z.real();
+	double piy = M_PI * z.imag();
+	double abspiy = std::fabs(piy);
+	double sinpix = cephes::sinpi(x);
+	double cospix = cephes::cospi(x);
+
+	if (abspiy < 700) {
+	    return std::complex<double>(sinpix*std::cosh(piy),
+					cospix*std::sinh(piy));
+	}
```

Comment:
I'd encourage us to do `return {sinpix*std::cosh(piy), cospix*std::sinh(piy)}`. That is shorthand for what you are doing here, it calls the return type constructor with those parameters.